src/util/conditionOperators/index.js

```diff
@@ -192,6 +192,15 @@ const filterComponentsForConditionComponentFieldOptions = (flattenedComponents)
         'button',
       ].includes(component.type);
 
+      // Change type for Day component with hidden fields to not treat it as a date
+      if (component.type === 'day' && (
+        component.fields.day.hide ||
+        component.fields.month.hide ||
+        component.fields.year.hide)
+      ) {
+        component.type = 'component';
+      }
```

Comment:
After this change, the date value will be handled as a string, not a date. Some operators, like greaterThen, will not work in this case. I believe there should be other solution.

Reply:
@johnformio, if we want to handle a day component value as a string, we need to make some operators unavailable for it, like greaterThan, lessThan and some other.
But I do not understand why we should do it. 'DD/YYYY' or 'DD/MM' are  valid date formats for moment js that means we can handle those formats as dates and apply all available date operators as before.
![image](https://github.com/user-attachments/assets/c62b8fe1-f6f6-4970-b571-a107a9ff51b8)